src/Permission.ts

```diff
@@ -0,0 +1,11 @@
+enum DrawPermission {
+  All,
+  Slave,
+  Master,
```

Comment:
Why not Allowed and Forbidden? Any particular reason for "Slave" and "Master" ? Knowing that Maste ris not used at all, and All doesn't differ from Master.

Reply:
In reality there is 3 state and it could be used in the future, or by people who use the engine, so i found cleaner to explicitly declare the different states.

Reply:
Not convinced right now, we're gonna take this over to a phone call!

Reply:
Okay, convinced now! Thanks!